### PR TITLE
remove redundant GetTypeInfo

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Review by Evangelink — COMMENTED** [review]
@nohwnd @MarcoRossignoli Could I get a second review please?

@SimonCropp it seems that there are some unused directives causing the build failure.

**Review by MarcoRossignoli — APPROVED** [review]
LGTM with green build/tests

**Review by Evangelink — APPROVED** [review]

**Review by Evangelink — APPROVED** [review]

**Comment by SimonCropp** [comment]
any ideas on what the remaining errors are about?

**Pull request merged**
